services/rds/src/main/java/software/amazon/awssdk/services/rds/DefaultRdsUtilities.java

```diff
@@ -0,0 +1,141 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.rds;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.signer.Aws4Signer;
+import software.amazon.awssdk.auth.signer.params.Aws4PresignerParams;
+import software.amazon.awssdk.awscore.client.config.AwsClientOption;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.rds.model.GenerateAuthenticationTokenRequest;
+import software.amazon.awssdk.utils.StringUtils;
+
+@Immutable
+@SdkInternalApi
+final class DefaultRdsUtilities implements RdsUtilities {
+    private final Region region;
+    private final AwsCredentialsProvider credentialsProvider;
+
+    DefaultRdsUtilities(DefaultBuilder builder) {
+        this.credentialsProvider = builder.credentialsProvider;
+        this.region = builder.region;
+    }
+
+    /**
+     * Used by RDS low-level client's utilities() method
+     */
+    @SdkInternalApi
+    static RdsUtilities create(SdkClientConfiguration clientConfiguration) {
+        return new DefaultBuilder().clientConfiguration(clientConfiguration).build();
+    }
+
+    @Override
+    public String generateAuthenticationToken(GenerateAuthenticationTokenRequest request) {
+        SdkHttpFullRequest httpRequest = SdkHttpFullRequest.builder()
+                                            .method(SdkHttpMethod.GET)
+                                            .protocol("https")
+                                            .host(request.hostname())
+                                            .port(request.port())
+                                            .encodedPath("/")
+                                            .putRawQueryParameter("DBUser", request.userName())
+                                            .putRawQueryParameter("Action", "connect")
+                                            .build();
+
+        Instant expirationTime = Instant.now().plus(15, ChronoUnit.MINUTES);
+        Aws4PresignerParams presignRequest = Aws4PresignerParams.builder()
+                                                .expirationTime(expirationTime)
+                                                .awsCredentials(resolveCredentials(request).resolveCredentials())
+                                                .signingName("rds-db")
+                                                .signingRegion(resolveRegion(request))
+                                                .build();
+
+        SdkHttpFullRequest fullRequest = Aws4Signer.create().presign(httpRequest, presignRequest);
```

Comment:
Seems like we can reuse the signer.

Reply:
Ok, it lacked the thread safe annotation so wasnt sure.